macOS Support

doctashay opened the pull request:
Adds initial, incomplete support for building on macOS. Project builds but fails to open due to an OpenGL context issue. Will be addressed

doctashay committed:
Fix building issues for macOS

Addresses compiler-specific issues preventing the build from completing on macOS. 

Currently builds, but fails to create OpenGL context due to OS-specific issue

doctashay committed:
Merge remote-tracking branch 'origin/documentation-update' into macos-support

detolly commented:
Has been mentioned in DMs, but I'll mention it here if anyone wants to add onto this or look into it:

Apple seems to support legacy profile opengl only. The shaders may or may not use some features that are only present in newer versions. The fix is to ifdef static initialization of Object shaders, that are present in the Renderer/Object.cpp file, and have custom shaders for macos. In addition to this, the imgui impl has to be changed.

doctashay committed:
Additional fixes for GLFW on macOS 10.15 and 11

doctashay committed:
Additional fixes for GLFW on macOS 10.15 and 11 

Still having additional issues creating a GLFW context on Catalina and Big Sur machines, although this may be an issue with my KVM and graphics acceleration. Regardless, any OpenGL contexts on these platforms will be limited to software mode.

doctashay committed:
Merge remote-tracking branch 'origin/macos-support' into macos-support

doctashay committed:
Revert "Merge branch 'macos-support'"

This reverts commit bfa36111e9d5e9672e84380a942d371bf69e64e8.
